spring-jdbc/src/main/java/io/crate/spring/jdbc/samples/da/PostgresJDBCDBObjectsFactory.java

```diff
@@ -0,0 +1,45 @@
+package io.crate.spring.jdbc.samples.da;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.dao.DataIntegrityViolationException;
+
+import io.crate.shade.org.postgresql.util.PGobject;
+import io.crate.spring.jdbc.samples.domain.BlogPost;
+
+public class PostgresJDBCDBObjectsFactory implements JDBCDBObjectsFactory {
+
+    private ObjectMapper mapper = new ObjectMapper();
+    
+    @Override
+    public Object createUserObject(BlogPost post) {
+        
+        if (post != null && post.getUser() != null) {
+            // objects can be streamed as json strings,
+            // https://crate.io/docs/reference/en/latest/protocols/postgres.html#jdbc
+            PGobject userObject = new PGobject();
+            userObject.setType("json");
+            Map<String, Object> userValue = new HashMap<String, Object>();
```

Comment:
```suggestion
            HashMap<String, Object> userValue = new HashMap<String, Object>();
```

Reply:
Really. Why? Implementation of Map is not interesting at all

Reply:
There is no benefit of programming inside a method against an interface, and besides that it is actually less efficient since there is one additional lookup.

Reply:
There is no benefit on programming against an interface in a method, but ok.

Reply:
> Really. Why? Implementation of Map is not interesting at all

Using the interface results in different bytecode:

```
INVOKEINTERFACE java/util/Map.put (Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object; (itf)
```

Compared to programming against the `HashMap`:

```
INVOKEVIRTUAL java/util/HashMap.put (Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
```

`INVOKEINTERFACE` is slower than the `INVOKEVIRTUAL`. That's why we tend to default to using the concrete implementations within a local scope. Doesn't matter for the sample app and the suggestion probably could've been omitted, but it's a good habit to pick up and we're used to point out such things.

In some cases using the new `var` keyword can also be an option for that reason - unless it degrades readability.
(Leaking implementation details across methods or class boundaries is usually still a bad idea).